src/mlpack/tests/lars_test.cpp

```diff
@@ -371,30 +371,53 @@ BOOST_AUTO_TEST_CASE(LARSTrainReturnCorrelation)
   // Test with Cholesky decomposition and with lasso.
   LARS lars1(true, lambda1, 0.0);
   arma::vec betaOpt1;
-  double maxCorr = lars1.Train(X, y, betaOpt1);
+  double error = lars1.Train(X, y, betaOpt1);
 
-  BOOST_REQUIRE_EQUAL(std::isfinite(maxCorr), true);
+  BOOST_REQUIRE_EQUAL(std::isfinite(error), true);
 
   // Test without Cholesky decomposition and with lasso.
   LARS lars2(false, lambda1, 0.0);
   arma::vec betaOpt2;
-  maxCorr = lars2.Train(X, y, betaOpt2);
+  error = lars2.Train(X, y, betaOpt2);
 
-  BOOST_REQUIRE_EQUAL(std::isfinite(maxCorr), true);
+  BOOST_REQUIRE_EQUAL(std::isfinite(error), true);
 
   // Test with Cholesky decomposition and with elasticnet.
   LARS lars3(true, lambda1, lambda2);
   arma::vec betaOpt3;
-  maxCorr = lars3.Train(X, y, betaOpt3);
+  error = lars3.Train(X, y, betaOpt3);
 
-  BOOST_REQUIRE_EQUAL(std::isfinite(maxCorr), true);
+  BOOST_REQUIRE_EQUAL(std::isfinite(error), true);
 
   // Test without Cholesky decomposition and with elasticnet.
   LARS lars4(false, lambda1, lambda2);
   arma::vec betaOpt4;
-  maxCorr = lars4.Train(X, y, betaOpt4);
+  error = lars4.Train(X, y, betaOpt4);
 
-  BOOST_REQUIRE_EQUAL(std::isfinite(maxCorr), true);
+  BOOST_REQUIRE_EQUAL(std::isfinite(error), true);
+}
+
+/**
+ * Test that LARS::ComputeError() returns error value less than 1
+ * and greater than 0.
+ */
+BOOST_AUTO_TEST_CASE(LARSTestComputeError)
+{
+  arma::mat X;
+  arma::mat Y;
+
+  data::Load("lars_dependent_x.csv", X);
+  data::Load("lars_dependent_y.csv", Y);
+
+  arma::rowvec y = Y.row(0);
+
+  LARS lars1(true, 0.1, 0.0);
+  arma::vec betaOpt1;
+  lars1.Train(X, y, betaOpt1);
+  double cost = lars1.ComputeError(X, y);
```

Comment:
We know that the Train() function also returns the same error as ComputeError. Can you verify(By adding a BOOST_EQUAL assertion) that the error returned by `Train()` and `ComputeError` is the same?.

Reply:
Thanks @walragatver for the review added an assertion for that.